cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -96,6 +97,46 @@ def _validate(self, os, scheduler):
             )
 
 
+class CustomAmiTagValidator(Validator):
+    """Custom AMI tag validator to check if the AMI was created by pcluster to avoid runtime baking."""
+
+    def _validate(self, custom_ami: str):
+        tags = AWSApi.instance().ec2.describe_image(custom_ami).tags
+        tags_dict = {}
+        if tags:  # tags can be None if there is no tag
+            for tag in tags:
+                tags_dict[tag["Key"]] = tag["Value"]
+        if PCLUSTER_VERSION_TAG not in tags_dict:
+            self._add_failure(
+                (
+                    "The custom AMI may not have been created by pcluster."
+                    "You can ignore this warning if the AMI is shared or copied from another pcluster AMI."
+                    "If the AMI is indeed not created by pcluster, cluster creation will fail."
+                    "If the cluster creation fails, please see the cloud-init-output.log on the head node"
+                    " to confirm the root cause."
+                ),
+                FailureLevel.ERROR,
```

Comment:
I think this can be a WARN but the other one on which the verison is different an ERROR